Mathlib/Topology/Constructions.lean

```diff
@@ -861,6 +861,15 @@ theorem embedding_prod_mk (x : X) : Embedding (Prod.mk x : Y → X × Y) :=
 
 end Prod
 
+section Bool
+
+lemma continuous_bool_rng [TopologicalSpace X] {f : X → Bool} :
+    Continuous f ↔ IsClopen (f ⁻¹' {true}) := by
```

Comment:
What do you think of adding also the analogous lemma with `false`?

Reply:
Is it useful?

Reply:
I am not sure if it will be useful and I do not want to push for an extra lemma either.  I'll leave a further suggestion, but will not mention this anymore!
```lean
lemma continuous_bool_rng [TopologicalSpace X] {f : X → Bool} (b : Bool) :
    Continuous f ↔ IsClopen (f ⁻¹' {b}) := by
  cases b <;>
    rw [continuous_discrete_rng, Bool.forall_bool, IsClopen, ← isOpen_compl_iff, ← preimage_compl,
      Bool.compl_singleton]
  · rw [Bool.not_false, And.comm]
  · rw [Bool.not_true]
```

Reply:
The right way to do it is to add `Bool.forall_bool' {p : Bool → Prop} (b : Bool) : (∀ x, p x) ↔ p b ∧ p !b` and use it here. I'll do it tomorrow.